### DISCUSSION TITLE
Permission issues on the host, entrypoint.sh cannot make directories or files.

### DISCUSSION THREAD

**Discussion by barrettsmits** [felddy/foundryvtt-docker#298] [Q&A] [answered]
I am having a very hard time getting this to work "out of the box"
I get the following error on  docker-compose up
./entrypoint.sh: line 210: can't create /data/Config/license.json: Permission denied

I get an error with mkdir no permissions when the Config folder is not pre-created. 

I have tried the following: 
granted docker group permissions. 
Granted 777 permissions.
reinstalled OS
tried an another docker-compose, worked, but didn't make any specific directories on the host. 

I Initially placed in /opt/foundry/
tried /tmp/foundry and a new root folder called /proj/ (just to be 100% permissions where correct. )

I'm at my wits end so I'm looking to see if anyone has a suggestion. 

I'm running Fedora 35, fresh install with Docker and docker-compose installed. 

I have my user, docker group, and have added my user to the docker group.

I've followed post setup of docker to be sure the permissions where there, other containers work,

Reviewing the src\entrypoint.sh file. I'm seeing that  entrypoint,sh doesn't have any uid or gid assignments, is that intended? I saw a previous discussion about supplying the uid and gid in Env, but that doesn't seem to help with my issue. (Because those are for inside the container.)

I'm 100% sure this is an issue with my environment. any help would be appreciated.

**Comment by felddy** [comment, marked as answer]
Could you share your `docker-compose.yml` file and the container logs in verbose mode.  
_If you are passing credentials through environment variables be sure the sanitize them out._

Off the top of my head:  are you running [SELinux](https://docs.fedoraproject.org/en-US/quick-docs/changing-selinux-states-and-modes/)?

**Reply by barrettsmits** [reply]
I turned off SElinux and tested, all issues I was having went away. 
Compose file was copy of this repos, with secrets file.

Thanks for the response and easy initial solution, I'll see what I need to do to reenable SElinux now that I undersstand that is the root of the issues I was seeing. 

Thank you.

**Reply by barrettsmits** [reply]
Follow-up on this, 

All was working when I turned off SElinux, however I want a secure workstation, so I worked through a few steps and managed to get this to work* (not fully tested) with the following setup:


run command: 
chcon -R -t container_file_t /PATH/TO/HOST/DIR


sources:
- https://github.com/containers/podman/issues/10779 
- https://www.arhea.net/posts/2020-04-28-selinux-for-containers.html
